### PR TITLE
Add segment

### PR DESCRIPTION
**Pull request by yields**
Add [Segment](https://segment.com), closes https://github.com/ampproject/amphtml/issues/3418.

```html
<amp-analytics type="segment">
<script type="application/json">
  {
    "vars": {
      "writeKey": "<key>",
    }
  }
...
```

### CODE REVIEW

File: extensions/amp-analytics/0.1/vendors.js [hunk 1]
```diff
@@ -622,6 +622,37 @@ export const ANALYTICS_CONFIG = {
     },
   },
 
+  'segment': {
```
Comment (cramforce) [inline extensions/amp-analytics/0.1/vendors.js]: We automatically generate a golden file test for this that you will need to update. See https://travis-ci.org/ampproject/amphtml/builds/138287933

**Comment by avimehta** [comment]
lgtm. Thanks for adding the support!

**Comment by jridgewell** [comment]
@yields: Have you signed the CLA yet?

**Comment by calvinfo** [comment]
We're just double checking with the lawyer that it's all good, but we should be ready to go shortly! :)

**Comment by yields** [comment]
@jridgewell signed and re-based!

**Pull request merged**